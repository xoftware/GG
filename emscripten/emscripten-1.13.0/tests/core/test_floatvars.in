#include <stdio.h>

// headers test, see issue #1013
#include <cfloat>
#include <cmath>

int main(int argc, char **argv) {
  float x = 1.234, y = 3.5, q = 0.00000001;
  y *= 3;
  int z = x < y;
  printf("*%d,%d,%.1f,%d,%.4f,%.2f*\n", z, int(y), y, (int)x, x, q);

  printf("%.2f, %.2f, %.2f, %.2f\n", fmin(0.5, 3.3), fmin(NAN, 3.3),
         fmax(0.5, 3.3), fmax(NAN, 3.3));

  printf("small: %.10f\n", argc * 0.000001);

  /*
  // Rounding behavior
  float fs[6] = { -2.75, -2.50, -2.25, 2.25, 2.50, 2.75 };
  double ds[6] = { -2.75, -2.50, -2.25, 2.25, 2.50, 2.75 };
  for (int i = 0; i < 6; i++)
    printf("*int(%.2f)=%d,%d*\n", fs[i], int(fs[i]), int(ds[i]));
  */

  return 0;
}
