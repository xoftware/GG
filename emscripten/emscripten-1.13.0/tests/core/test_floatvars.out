*1,10,10.5,1,1.2340,0.00*
0.50, 3.30, 3.30, 3.30
small: 0.0000010000
